File: backend/utils/items.py
# Items Catalog - Structured inventory for the Online Delivery System

ITEMS_CATALOG = {
    "Grocery": [
        {"name": "White Bread", "price": 20, "unit": "piece"},
        {"name": "Bhuja Chana", "price": 100, "unit": "kg"},
        {"name": "Maggie", "price": [15, 56], "unit": "packet"},
        {"name": "Moongfali", "price": 160, "unit": "kg"},
        {"name": "Soyabean (Packet)", "price": 10, "unit": "packet"},
        {"name": "Soyabean (Open - Small)", "price": 100, "unit": "kg"},
        {"name": "Macaroni / Pasta", "price": 60, "unit": "kg"},
        {"name": "Sugar (Chini)", "price": 45, "unit": "kg"},
        {"name": "Glucose D", "price": [56, 89], "unit": "packet"},
        {"name": "Jam (Kissan Mixed Fruit)", "price": [25, 90], "unit": "bottle"},
        {"name": "Colgate Toothpaste", "price": [20, 70, 130], "unit": "piece"},
        {"name": "Closeup Toothpaste", "price": [20, 55], "unit": "piece"},
        {"name": "Sensodyne Toothpaste", "price": 140, "unit": "piece"},
        {"name": "Coffee (Nescafe)", "price": [10, 124, 235], "unit": "packet"},
        {"name": "Mustard Oil (Sarso Tel - Sudh Kolhu)", "price": 180, "unit": "liter"},
        {"name": "Fortune Oil", "price": 190, "unit": "liter"},
        {"name": "Garam Masala (Whole)", "price": 20, "unit": "packet"},
        {"name": "Garam Masala (Packet)", "price": 10, "unit": "packet"},
        {"name": "Kacha Chana", "price": 80, "unit": "kg"},
        {"name": "Makhana", "price": 130, "unit": "100 gm"},
        {"name": "Kaju (Cashew)", "price": 1000, "unit": "kg"},
        {"name": "Badam (Almond)", "price": 1000, "unit": "kg"},
        {"name": "Pista (Packet)", "price": 350, "unit": "packet"},
    ],
    "Stationary": [
        {"name": "Practical File (Physics)", "price": [30, 60], "unit": "piece"},
        {"name": "Practical File (Chemistry)", "price": [30, 60], "unit": "piece"},
        {"name": "Practical File (General)", "price": [30, 60], "unit": "piece"},
        {"name": "Pen", "price": [4, 5, 10], "unit": "piece"},
        {"name": "Pencil", "price": 5, "unit": "piece"},
        {"name": "Eraser", "price": 5, "unit": "piece"},
        {"name": "Sharpener", "price": 5, "unit": "piece"},
        {"name": "Drawing Book (Engineering) A3 (36 pages)", "price": 90, "unit": "piece"},
        {"name": "Drawing Book (Engineering) A3 (56 pages)", "price": 110, "unit": "piece"},
        {"name": "A4 Size Pages (500 sheets)", "price": 200, "unit": "pack"},
        {"name": "Stapler", "price": 50, "unit": "piece"},
        {"name": "Staple Pins", "price": 10, "unit": "box"},
        {"name": "Sticky Notes", "price": 20, "unit": "pack"},
        {"name": "Fevicol", "price": [10, 20, 50], "unit": "bottle"},
        {"name": "Double-Sided Tape", "price": [5, 10], "unit": "roll"},
        {"name": "Spiral Notebook (Plain)", "price": [80, 100, 140], "unit": "piece"},
        {"name": "Spiral Notebook (Lined)", "price": [80, 100, 140], "unit": "piece"},
        {"name": "Register (A4 Rough)", "price": [30, 40, 50], "unit": "piece"},
        {"name": "Register (A2)", "price": [30, 40, 50], "unit": "piece"},
        {"name": "Register (A1)", "price": [30, 40, 50], "unit": "piece"},
        {"name": "Graph Book", "price": 10, "unit": "piece"},
        {"name": "Cello Tape", "price": 5, "unit": "roll"},
        {"name": "Practical File Pages (Lined)", "price": 10, "unit": "pack"},
        {"name": "Practical File Pages (Plain)", "price": 10, "unit": "pack"},
        {"name": "Practical File Cover (Plastic)", "price": [5, 10], "unit": "piece"},
        {"name": "Practical File Cover (Cardboard)", "price": 20, "unit": "piece"},
        {"name": "Highlighter", "price": 20, "unit": "piece"},
    ],
    "Fruits": [
        {"name": "Banana", "price": 40, "unit": "kg"},
        {"name": "Pomegranate (Anaar)", "price": 180, "unit": "kg"},
        {"name": "Mango", "price": 100, "unit": "kg"},
        {"name": "Apple", "price": 200, "unit": "kg"},
        {"name": "Orange (Santara)", "price": 150, "unit": "kg"},
        {"name": "Papaya (Papita)", "price": 60, "unit": "kg"},
        {"name": "Kiwi", "price": 50, "unit": "piece"},
        {"name": "Coconut (Nariyal)", "price": 70, "unit": "piece"},
        {"name": "Grapes (Angoor)", "price": 150, "unit": "kg"},
    ],
}


def get_items_catalog() -> dict:
    """Return the items catalog."""
    return ITEMS_CATALOG


def get_all_items() -> list:
    """Return all items as a flat list with category."""
    all_items = []
    for category, items in ITEMS_CATALOG.items():
        for item in items:
            all_items.append({
                "category": category,
                "name": item["name"],
                "price": item["price"],
                "unit": item["unit"]
            })
    return all_items


def format_price(price) -> str:
    """Format price for display."""
    if isinstance(price, list):
        return f"₹{', ₹'.join(map(str, price))}"
    return f"₹{price}"

File: frontend/data/items.ts
const API_BASE = process.env.NEXT_PUBLIC_API_URL;

export interface MenuItem {
  name: string;
  price: number;
  unit: string;
}

export interface ItemsCatalog {
  [category: string]: MenuItem[];
}

export const itemsCatalogPromise: Promise<ItemsCatalog> = getItemsCatalog();

export async function getItemsCatalog(): Promise<ItemsCatalog> {
  try {
    const res = await fetch(`${API_BASE}/catalog`);
    const json = await res.json();
    return json.data || {};
  } catch (err) {
    console.error("Failed to fetch catalog", err);
    return {};
  }
}

export const ITEM_IMAGE_MAP: Record<string, string> = {
  'White Bread': 'white_bread_metro_gold.png',
  'Bhuja Chana': 'roasted_chana_250gm.jpg',
  'Maggie (Small - 75g)': 'maggie_small_packet.jpg',
  'Maggie (Family Pack - 400g)': 'maggie_family_pack.jpg',
  'Moongfali': 'moongfali_peanut_250gm.jpg',
  'Soyabean (Packet)': 'soyachunks_fortune_packet.jpg',
  'Soyabean (Open - Small)': 'soyabean_chunks_open_250gm.jpeg',
  'Macaroni / Pasta': 'fusli_pasta_microni_250gm.jpeg',
  'Sugar (Chini)': 'suger_chini_250gm.jpg',
  'Glucose D (200g)': 'glucose_d_small.jpg',
  'Glucose D (400g)': 'glucose_d_large.jpeg',
  'Jam (Small)': 'kissan_mixed_fruit_jam_small.jpeg',
  'Jam (Big)': 'kissan_mixed_fruit_jam_large.jpeg',
  'Colgate (Small)': 'colgate_toothpaste_small.jpg',
  'Colgate (Medium)': 'colgate_toothpaste_large.webp',
  'Colgate (Big)': 'colgate_toothpaste_large.webp',
  'Closeup (Small)': 'close_up_toothpaste.jpg',
  'Closeup (Big)': 'close_up_toothpaste.jpg',
  'Sensodyne Toothpaste': 'sensodine_toothpaste.jpeg',
  'Coffee (Nescafe - 50g)': 'nescafe_coffee_small_packet.webp',
  'Coffee (Nescafe - 100g)': 'nescafe_coffee_medium.jpg',
  'Coffee (Nescafe - 200g)': 'nescafe_coffee_large.jpeg',
  'Mustard Oil (Sarso Tel)': 'sarso_tel_sudhh_ghani_1kg.jpg',
  'Fortune Oil': 'sarso_tel_fortune_1kg.jpg',
  'Garam Masala (Whole)': 'khada_garam_masala_packet.jpeg',
  'Garam Masala (Packet)': 'khada_garam_masala_packet.jpeg',
  'Kacha Chana': 'kacha_chana_250gm.jpg',
  'Makhana': 'makhana_100gm.jpeg',
  'Kaju (Cashew)': 'kaju_100gm.jpg',
  'Pista (Packet)': 'fresh_ pista_packet.jpg',
  'Badam (Almond)': 'badam_100gm.jpg',
  'Lassi (250ml)': 'amul_lassi_250ml_tetrapack_24rupees.jpeg',
  'Buttermilk (200ml)': 'amul_masti_spiced_buttermilk_200ml_14rupees.jpg',
  'Buttermilk (1L)': 'amul_masti_spiced_buttermilk_1lt_65rupees.jpg',
  'Practical File (Physics - 30 Pages)': 'practical_file_physics.jpg',
  'Practical File (Physics - 60 Pages)': 'practical_file_physics.jpg',
  'Practical File (Chemistry - 30 Pages)': 'practical_file_chemistry.jpg',
  'Practical File (Chemistry - 60 Pages)': 'practical_file_chemistry.jpg',
  'Practical File (General - 30 Pages)': 'practical_file_general.jpeg',
  'Practical File (General - 60 Pages)': 'practical_file_general.jpeg',
  'Drawing Book (A3 - 36 pages)': 'engineering_drawing_36page.jpg',
  'Drawing Book (A3 - 56 pages)': 'engineering_drawing_56page.jpg',
  'Pen (Blue, Cheap)': 'elkos_pen_4rupees.jpg',
  'Pen (Black, Cheap)': 'elkos_pen_4rupees.jpg',
  'Pen (Blue, Medium)': 'elkos_pen_5rupees.jpg',
  'Pen (Black, Medium)': 'elkos_pen_5rupees.jpg',
  'Pen (Blue, Good)': 'elkos_pen_10rupees.jpg',
  'Pen (Black, Good)': 'elkos_pen_10rupees.jpg',
  'Pencil': 'natraj_pencil.png',
  'Eraser': 'eraser_apsara.jpg',
  'Sharpener': 'pencil_sharpner_plastic.jpg',
  'A4 Size Pages (500 sheets)': 'a4_size_paper_500pages.jpg',
  'Stapler': 'stapler.jpg',
  'Staple Pins': 'staples_pin_box.jpeg',
  'Sticky Notes': 'sticky_notes_small_packet.jpeg',
  'Fevicol (Small)': 'fevicol_small.jpg',
  'Fevicol (Medium)': 'fevicol_small.jpg',
  'Fevicol (Big)': 'fevicol_small.jpg',
  'Double-Sided Tape (Small)': 'cello_tape_small.jpeg',
  'Double-Sided Tape (Big)': 'cello_tape_small.jpeg',
  'Spiral Notebook (Plain - 100 Pages)': 'spiral_notebook_100pg.jpg',
  'Spiral Notebook (Plain - 200 Pages)': 'spiral_notebook_200pg.jpeg',
  'Spiral Notebook (Lined - 100 Pages)': 'spiral_notebook_100pg.jpg',
  'Register (A4 Rough - 100 Pages)': 'register_a4_rough.jpg',
  'Register (A4 Rough - 200 Pages)': 'register_a4_rough_a4.jpg',
  'Graph Book': 'graph_book.jpg',
  'Cello Tape': 'cello_tape_small.jpeg',
  'Highlighter': 'highlighter.jpeg',
  'Banana': 'fresh_bannana_500gm.webp',
  'Pomegranate (Anaar)': 'pomegranate_anar_500gm.jpg',
  'Mango': 'mango_aam_500gm.jpg',
  'Apple': 'fresh_apple_500gm.jpg',
  // 'Orange (Santara)': 'orange_santara_500gm.jpg',
  'Papaya (Papita)': 'fresh_papaya_1kg.jpg',
  'Kiwi': 'fresh_kiwi_1piece.jpg',
  'Coconut (Nariyal)': 'nariyal_water.webp',
  'Grapes (Angoor)': 'fresh_angoor_250gm.jpg',
  'Eggs (30 pieces)': 'eggs_tray_30_piece_180rupees.jpg',
  'Raw Chicken (500g)': 'raw_chicken_500gm_120rupees.jpg',
};

export const DELIVERY_CHARGE = parseInt(process.env.NEXT_PUBLIC_DELIVERY_CHARGE); // Default to 9 if not set or invalid

export const HOSTEL_OPTIONS = [
  { value: 'BH1', label: 'BH1 (Boys Hostel 1)' },
  { value: 'BH2', label: 'BH2 (Boys Hostel 2)' },
  { value: 'BH3', label: 'BH3 (Boys Hostel 3)' },
  { value: 'GH1', label: 'GH1 (Girls Hostel)' },
];